### DISCUSSION TITLE
[ERR_REQUIRE_ESM] How to build SSR Quasar project with Openlayers?

### DISCUSSION THREAD

**Discussion by sergeykorobeynikov** [quasarframework/quasar#15202] [CLI - SSR mode]
I use Openlayers in my project, for example StartPage.vue contains the following code:

```vue
<script lang="ts">
//...
import { Map, View } from 'ol';
import { Tile as TileLayer } from 'ol/layer';
import { fromLonLat } from 'ol/proj';
import XYZ from 'ol/source/XYZ';
//...
</script>
```

When I start a dev server for my App using "**quasar dev -m ssr**" cmd everything starts(!) and there are no problems(!).

When I build SSR project using "**quasar build -m ssr**" cmd and, after that run project:
```bat
cd ./dist/ssr
npm start
```
I see:
> Error [ERR_REQUIRE_ESM]: require() of ES Module C:\dev\proj1\node_modules\ol\index.js from C:\dev\proj1\dist\ssr\server\server-entry.js not supported.      
> Instead change the require of index.js in C:\dev\proj1\dist\ssr\server\server-entry.js to a dynamic import() which is available in all CommonJS modules.        
>     at Module.<anonymous> (C:\dev\proj1\dist\ssr\server\server-entry.js:1:261)
>     at Object.<anonymous> (C:\dev\proj1\dist\ssr\index.js:1:1269) {
>   code: 'ERR_REQUIRE_ESM'
> }
> 
> Node.js v18.12.1

What is needed to solve this problem?

P.S. I am using ol@7.1.0.

**Comment by 209** [comment]
bump

**Comment by rtcpw** [comment]
This seems to be the same issue as this:
https://github.com/quasarframework/quasar/discussions/13726
